In [3]:
from google.colab import drive

drive.mount('/content/drive')

Mounted at /content/drive


In [16]:
%cd drive/Othercomputers/My\ MacBook\ Pro/GPU-Computing/

[Errno 2] No such file or directory: 'drive/Othercomputers/My MacBook Pro/GPU-Computing/'
/content/drive/Othercomputers/My MacBook Pro/GPU-Computing


In [17]:
!pwd

/content/drive/Othercomputers/My MacBook Pro/GPU-Computing


In [18]:
!which nvcc

/usr/local/cuda/bin/nvcc


In [19]:
!g++ -std=c++17 ./serial_image_blur.cpp -o serial-blur && ./serial-blur

elapsed time CPU 40.7013


In [44]:
!nvcc -arch=sm_75 ./profiling.cu -o vec_profile 

In [28]:
!pwd

/content/drive/Othercomputers/My MacBook Pro/GPU-Computing


In [42]:
!ls 


apple.jpg				    RGB_to_gray_serial.cpp
main					    run.ipynb
parallel_blur.png			    serial-blur
parallel_gray.png			    serial_blur.png
parallel_image_blur.cu			    serial_gray.png
parallel_matrix_mul.cu			    serial_image_blur.cpp
parallel_matrix_mul_tiling_benchmarking.cu  stb_image.h
parallel_matrix_mul_tiling.cu		    stb_image_write.h
parallel_matrix_mul_tiling_opt.cu	    timeline.prof
profiling.cu				    vec_profile
README.md				    vector_addition.cu
RGB_to_gray_parallel.cu


In [53]:
!nvprof -f -o timeline.prof ./vec_profile

==18061== NVPROF is profiling process 18061, command: ./vec_profile
elapsed time CPU 1179.69
elapsed time GPU 9.95544
==18061== Generated result file: /content/drive/Othercomputers/My MacBook Pro/GPU-Computing/timeline.prof


In [54]:
!nvprof -m all ./vec_profile

======== Warning: Skipping profiling on device 0 since profiling is not supported on devices with compute capability 7.5 and higher.
                  Use NVIDIA Nsight Compute for GPU profiling and NVIDIA Nsight Systems for GPU tracing and CPU sampling.
                  Refer https://developer.nvidia.com/tools-overview for more details.

==19985== NVPROF is profiling process 19985, command: ./vec_profile
elapsed time CPU 1185.37
elapsed time GPU 9.65949
==19985== Profiling application: ./vec_profile
==19985== Profiling result:
No events/metrics were profiled.


======== Warning: Skipping profiling on device 0 since profiling is not supported on devices with compute capability 7.5 and higher.
                  Use NVIDIA Nsight Compute for GPU profiling and NVIDIA Nsight Systems for GPU tracing and CPU sampling.
                  Refer https://developer.nvidia.com/tools-overview for more details.

We might need NVIDIA Nsight (new profiling) for new devices